sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/implementation/CloudEventTracingPipelinePolicy.java

```diff
@@ -34,8 +35,17 @@ public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineN
         StringBuilder bodyStringBuilder = new StringBuilder();
         if (TracerProxy.isTracingEnabled() && contentType != null
             && Constants.CLOUD_EVENT_CONTENT_TYPE.equals(contentType.getValue())) {
-            return request.getBody().map(byteBuffer -> bodyStringBuilder.append(new String(byteBuffer.array(),
-                StandardCharsets.UTF_8)))
+            return request.getBody()
+                .map(byteBuffer -> {
+                    if (byteBuffer.hasArray()) {
+                        return bodyStringBuilder.append(new String(byteBuffer.array(),
+                            byteBuffer.arrayOffset() + byteBuffer.position(), byteBuffer.remaining(),
+                            StandardCharsets.UTF_8));
+                    } else {
+                        return bodyStringBuilder.append(new String(FluxUtil.byteBufferToArray(byteBuffer.duplicate()),
```

Comment:
nit: maybe we should move `byteBufferToArray` helper method to `CoreUtils` instead of having it in `FluxUtil`.